sdk/ruby/lib/chain/connection.rb

```diff
@@ -111,7 +111,10 @@ def _single_request(path, body)
           http.send "#{k}=", @opts[k]
         end
 
-        req = Net::HTTP::Post.new(@url.request_uri + path)
+        full_path = @url.request_uri.chomp('/')
+        full_path += (path[0] == '/') ? path : '/' + path
```

Comment:
I don't know if consecutive slashes are possible in this case, since `@url` is a parsed URI object.